backend/src/graphql/model/Decay.ts

```diff
@@ -1,20 +1,24 @@
 import { ObjectType, Field, Int } from 'type-graphql'
 import Decimal from 'decimal.js-light'
 
+interface DecayInterface {
+  balance: Decimal
+  decay: Decimal
+  roundedDecay: Decimal
+  start: Date | null
+  end: Date | null
+  duration: number | null
+}
+
 @ObjectType()
 export class Decay {
-  constructor(
-    balance: Decimal,
-    decay: Decimal,
-    start: Date | null,
-    end: Date | null,
-    duration: number | null,
-  ) {
-    this.balance = balance
-    this.decay = decay
-    this.start = start
-    this.end = end
-    this.duration = duration
+  constructor(data: DecayInterface) {
```

Comment:
you can make `{ balance: Deciaml, ... }`  curly brackets around the parameters to archive the same thing.